R/locale.R

```diff
@@ -0,0 +1,40 @@
+#' Locale used by stringr
+#'
+#' @description
+#' `stringr_locale()` returns a single string representing the default locale
+#' used by stringr in functions like [str_sort()] and [str_to_upper()]. The
+#' default locale is `"en"` (American English) on all platforms, but can be
+#' customized by setting the global option, `tidyverse.locale_collation`, to a
+#' locale found in [stringi::stri_locale_list()].
+#'
+#' We only recommend setting the global option if you don't expect your code
+#' to be shared across multiple computers. Otherwise, it is more reproducible
+#' to set the `locale` argument directly.
+#'
+#' @export
+#' @examples
+#' stringr_locale()
+#'
+#' withr::with_options(
+#'   list(tidyverse.locale_collation = "fr"),
+#'   stringr_locale()
+#' )
+stringr_locale <- function() {
+  locale <- peek_option("tidyverse.locale_collation")
```

Comment:
I just want to gut check this one more time. Are we sure that we want this to be `tidyverse.locale_collation` which is shared between dplyr and stringr? Would it be surprising for users who set this argument specifically while working on dplyr to then find that stringr is returning different results?

Something about entangling the two packages together through this option feels _slightly_ wrong to me. Like, I feel like there should be `stringr.locale` and `dplyr.locale`. Setting each one of those individually doesn't feel like too much work, and then you avoid the above mentioned confusion and you don't have to remap C->en_US_POSIX.

Reply:
Feels like it's worth collectively discussing this somewhere? (Maybe just in slack). It feels to me that we really want a unified notion of tidyverse locale, not one local per package.